borg/hashindex.pyx

```diff
@@ -155,12 +166,13 @@ cdef class NSKeyIterator:
 cdef class ChunkIndex(IndexBase):
 
     value_size = 12
+    max_ref = 2**30-1  # to be *extra* safe, from constants like EMPTY and DELETED
```

Comment:
Why not make this 2**31-2 to double the space? Also 2**32-2 should also work it appears since the refcount cannot be negative, so we can always go for the unsigned as the counter, so (unsigned) -1 is deleted, (unsigned)-2 is empty and (unsigned) -3 is "too popular".

This is likely not super critical due to a sheer repetition count, I guess.